From 5a3ac26edcc5a4f731e1687473363607a6c63289 Mon Sep 17 00:00:00 2001
From: Adithya R <gh0strider.2k18.reborn@gmail.com>
Date: Sat, 9 Sep 2023 08:23:45 +0530
Subject: [PATCH] gki_defconfig: Disable per-task stackprotector

This is automatically enabled on newer compilers, however it kills off
the __stack_chk_guard symbol which is required by modules.

ERROR: Differences between ksymtab and symbol list detected!
Symbols missing from ksymtab:
 - __stack_chk_guard

Signed-off-by: fjrXTR <fajarslebew31@gmail.com>
---
 arch/arm64/Kconfig               | 3 ++-
 arch/arm64/configs/gki_defconfig | 1 +
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/Kconfig b/arch/arm64/Kconfig
index 6292950fddd3..e546fe9e040f 100644
--- a/arch/arm64/Kconfig
+++ b/arch/arm64/Kconfig
@@ -1958,8 +1958,9 @@ config CC_HAVE_STACKPROTECTOR_SYSREG
 	def_bool $(cc-option,-mstack-protector-guard=sysreg -mstack-protector-guard-reg=sp_el0 -mstack-protector-guard-offset=0)
 
 config STACKPROTECTOR_PER_TASK
-	def_bool y
+	bool "Stack Protector per task"
 	depends on STACKPROTECTOR && CC_HAVE_STACKPROTECTOR_SYSREG
+	default y
 
 endmenu
 
diff --git a/arch/arm64/configs/gki_defconfig b/arch/arm64/configs/gki_defconfig
index 218110d68835..790639062ee4 100644
--- a/arch/arm64/configs/gki_defconfig
+++ b/arch/arm64/configs/gki_defconfig
@@ -65,6 +65,7 @@ CONFIG_SETEND_EMULATION=y
 CONFIG_RANDOMIZE_BASE=y
 # CONFIG_RANDOMIZE_MODULE_REGION_FULL is not set
 CONFIG_CMDLINE="stack_depot_disable=on kasan.stacktrace=off kvm-arm.mode=protected cgroup_disable=pressure cgroup.memory=nokmem noirqdebug"
+# CONFIG_STACKPROTECTOR_PER_TASK is not set
 CONFIG_CMDLINE_EXTEND=y
 # CONFIG_DMI is not set
 CONFIG_PM_WAKELOCKS=y
